pkg/cloud/services/network/secondarycidr.go

```diff
@@ -72,17 +82,16 @@ func (s *Service) disassociateSecondaryCidr() error {
 		return err
 	}
 
-	if len(vpcs.Vpcs) != 1 {
-		return errors.Errorf("VPC not found")
+	if err = isVPCPresent(vpcs); err != nil {
+		return err
 	}
 
 	existingAssociations := vpcs.Vpcs[0].CidrBlockAssociationSet
 	for _, existing := range existingAssociations {
-		if existing.CidrBlock == s.scope.SecondaryCidrBlock() {
-			_, err := s.EC2Client.DisassociateVpcCidrBlock(&ec2.DisassociateVpcCidrBlockInput{
+		if reflect.DeepEqual(existing.CidrBlock, s.scope.SecondaryCidrBlock()) {
```

Comment:
I'm wonder if we use [go-cmp](https://github.com/google/go-cmp) instead

Reply:
Actually we have many use cases where we are using `reflect.DeepEqual()`, shall we consider this as a separate issue, to see how feasible is go-cmp vs reflect?

Reply:
Yes, lets consider this a separate issue. There are some nice benefits to using go-cmp instead. Shall we discuss at next weeks office hours?

Reply:
Yeah sure @richardcase, meanwhile I will raise an issue for the same so that we can discuss  in office hours

Reply:
I liked the idea of providing custom equality functions.

Reply:
> I liked the idea of providing custom equality functions.

Same here @sedefsavas , thats why i like go-cmp and it using an `Equals` function if one is defined on the type.